product_margin_classification/models/product_template.py

```diff
@@ -66,6 +78,19 @@ def _compute_theoretical_multi_template(self):
             template.theoretical_difference = 0.0
             template.margin_state = 0.0
 
+    def _inverse_margin_classification_id(self):
+        variant_count = len(self.product_variant_ids)
+        if variant_count == 1:
+            self.product_variant_ids.margin_classification_id = (
+                self.margin_classification_id
+            )
+        elif variant_count == 0:
+            archived_variants = self.with_context(active_test=False).product_variant_ids
+            if len(archived_variants) == 1:
+                archived_variants.margin_classification_id = (
+                    self.margin_classification_id
+                )
```

Comment:
i’m not sure i understand the logic here. what are the reasons to treat archived products separately? why not simply assign the margin classification to all products, archived or not?

Reply:
You're right. I first applied the same algorithm as for the _set_barcode function.
I finally used the algorithm of _set_weight. Should be OK now.

thanks for your review.